src/login_param.rs

```diff
@@ -479,69 +481,147 @@ impl ConfiguredLoginParam {
             .unwrap_or_default();
 
         if let Some(provider) = provider {
-            let addr_localpart = if let Some(at) = addr.find('@') {
-                addr.split_at(at).0.to_string()
-            } else {
-                addr.to_string()
-            };
-            imap = provider
-                .server
-                .iter()
-                .filter_map(|server| {
-                    if server.protocol != Protocol::Imap {
-                        return None;
-                    }
-
-                    let Ok(security) = server.socket.try_into() else {
-                        return None;
-                    };
-
-                    Some(ConfiguredServerLoginParam {
-                        connection: ConnectionCandidate {
-                            host: server.hostname.to_string(),
-                            port: server.port,
-                            security,
-                        },
-                        user: if !mail_user.is_empty() {
-                            mail_user.clone()
+            let parsed_addr = EmailAddress::new(&addr).context("Bad email-address")?;
+            let addr_localpart = parsed_addr.local;
+
+            if provider.server.is_empty() {
+                let servers = vec![
+                    ServerParams {
+                        protocol: Protocol::Imap,
+                        hostname: context
+                            .get_config(Config::ConfiguredMailServer)
```

Comment:
Shouldn't `ConfiguredImapServers`/`ConfiguredSmtpServers` be used instead? `ConfiguredMailServer` is legacy and reset to None in `save_as_configured_params()`